### ISSUE TITLE
Dierdre and Death Rider armor

### ISSUE THREAD

**Issue by cigumo** [cigumo/krli#1507]
http://support.ironhidegames.com/a/tickets/72856

> When using hero Deirdre’s ability “wings of light” to increase ally armor on the necromancer tower’s death rider, after using wings of light on the death rider once, then upgrading the death rider ability to level two or three, the wings of light no longer upgrades the death riders armor rating. Example, If you use wings of light, first on a level 1 death rider, then upgrade the death rider to level two, the armor rating does not increase from medium to high, but if if you upgrade the death rider before using Dierdre’s wings of light ability, the armor is increased from medium to high.

**Comment by Mmlh1** [comment]
This is true for everything, right? Upgrading removes buffs. That applies to all barracks when paired with Wings of Light, but it also applies to Dierdre's Consecration and Kahz's Roar of Fury (upgrading a tower removes the buff).

I do agree that it would be much nicer if upgrading would preserve buffs, but it's not specific to Death Rider.
